### PR TITLE
Update mamba to 0.27.0 and add checks for boa

### PR DESCRIPTION
**Pull request by hmaarrfk**
Checklist
* [ ] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [ ] Bumped the build number (if the version is unchanged)
* [ ] Reset the build number to `0` (if the version changed)
* [ ] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [ ] Ensured the license file is being packaged.

### CODE REVIEW

**Comment by hmaarrfk** [comment]
I don't really know how to make sense of these failures.

**Comment by hmaarrfk** [comment]
clap! should we try to update conda as well to 22.9?

File: scripts/test.sh [hunk 1]
```diff
@@ -53,6 +65,20 @@ else
   echo "***** Print conda info *****"
   conda info
   conda list
+  conda clean -yi
```
Comment (hmaarrfk) [inline scripts/test.sh:68]:
```suggestion
  conda clean --yes --index-cache
```
Just kicking off new builds.

File: scripts/test.sh [hunk 2]
```diff
@@ -53,6 +65,20 @@ else
...
+
+  if [[ "${INSTALLER_NAME}" == "Mambaforge" ]]; then
+    echo "***** Mambaforge detected. Checking for boa compatibility *****"
+    mamba_version_start=$(mamba --version | grep mamba | cut -d ' ' -f 2)
+    mamba info
+    mamba install boa --yes
```
Comment (hmaarrfk) [inline scripts/test.sh:74]:
```suggestion
    mamba install "mamba=${mamba_version_start}" boa --yes
```

**Comment by hmaarrfk** [comment]
Thank you @isuruf !

File: scripts/test.sh [hunk 3]
```diff
@@ -53,6 +65,20 @@ else
   echo "***** Print conda info *****"
   conda info
   conda list
+  conda clean --yes --index-cache
```
Comment (isuruf) [inline scripts/test.sh:68]: @hmaarrfk, this was just a debugging step. We should not release a miniforge with this requirement. This appears to be a bug in micromamba/mamba

Comment (hmaarrfk) [reply scripts/test.sh:68]: ah i misunderstood. I htink that @jakirkham and @ocefpaf might have gotten to the bottom of it.

Comment (hmaarrfk) [reply scripts/test.sh:68]: https://github.com/conda-forge/miniforge/pull/360

**Pull request merged**